internal/app/init/main.go

```diff
@@ -130,6 +131,16 @@ func initram() (err error) {
 	if err = kspp.EnforceKSPPKernelParameters(); err != nil {
 		return err
 	}
+	// Setup hostname if provided
+	if kargs, err := kernel.ParseProcCmdline(); err != nil {
```

Comment:
Since we are using this in two places, can we expose a function in `internal/pkg/net/net.go`? Something like `HostnameFromKernelParameter() (string, bool)`. We could then do something like

```
if hostname, ok := net.HostnameFromKernelParameter() {
    ...
}
```

There is the `error` to check in `ParseProcCmdline`, but I'd be fine with returning `"", false"` in that case.

Reply:
Actually, I was just leaving a comment about that, but I was going to ask if a more generic `GetKernelParameter(string) (string, bool)` would be useful.

Reply:
Sounds like a good idea to me!

Reply:
In that case. I'm happy to merge this and follow up with a PR implementing the above. Your call.